kernel/typing.ml

```diff
@@ -390,7 +392,31 @@ let subst_context (sub:SS.t) (ctx:typed_context) : typed_context =
   let apply_subst i (l,x,ty) = (l,x,Subst.apply_subst (SS.subst2 sub i) 0 ty) in
   List.mapi apply_subst ctx
 
-let check_rule sg (rule:untyped_rule) : SS.t * typed_rule =
+let check_type_annotations sg (sub:SS.t) (pctx:(loc * ident * typ) list) (annot:term option context) =
```

Comment:
I do think that this is really bad to have this in the kernel. This is more code to be trusted while I think it should not be trusted.